smac/utils/validate.py

```diff
@@ -237,31 +244,35 @@ def validate_epm(self, config_mode:Union[str, typing.List[Configuration]]='def',
             number of repetitions in nondeterministic algorithms
         runhistory: RunHistory
             optional, RunHistory-object to reuse runs
+        reuse_epm: bool
+            if true (and if self.epm), reuse epm to validate runs
 
         Returns
         -------
         runhistory: RunHistory
             runhistory with predicted runs
         """
-        if not isinstance(runhistory, RunHistory):
+        if not isinstance(runhistory, RunHistory) and (self.epm == None or
+                                                       reuse_epm == False):
             raise ValueError("No runhistory specified for validating with EPM!")
-        # Train random forest and transform training data (from given rh)
-        rh2epm = RunHistory2EPM4Cost(num_params=len(self.scen.cs.get_hyperparameters()),
-                                     scenario=self.scen, rng=self.rng)
-        X, y = rh2epm.transform(runhistory)
-        self.logger.debug("Training model with data of shape X: %s, y:%s",
-                          str(X.shape), str(y.shape))
-
-        types, bounds = get_types(self.scen.cs, self.scen.feature_array)
-        model = RandomForestWithInstances(types=types,
-                                          bounds=bounds,
-                                          instance_features=self.scen.feature_array,
-                                          seed=self.rng.randint(MAXINT),
-                                          ratio_features=1.0)
-        model.train(X, y)
+        elif reuse_epm == False or self.epm == None:
+            # Train random forest and transform training data (from given rh)
+            rh2epm = RunHistory2EPM4Cost(num_params=len(self.scen.cs.get_hyperparameters()),
+                                         scenario=self.scen, rng=self.rng)
```

Comment:
If we have runtime data and adaptive capping was used, you should set 
`impute_censored_data=True` and pass a `imputor` instance
to handle the censored data.